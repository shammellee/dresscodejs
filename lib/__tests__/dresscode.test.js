'use strict';

beforeAll(() =>
{
  global.Dresscode = require('../dresscode');

  var _style       = document.head.appendChild(document.createElement('style'))
      ,_stylesheet = _style.sheet
      ;

  _style.id = 'dresscode';

  document.body.innerHTML =
    '<div data-dresscode="my-class" '
    + 'data-padding="5px" '
    + 'class="'
    + 'background-color_blue '
    + 'color_yellow '
    + 'font-size_2rem '
    + 'padding_var-padding '
    + '"></div>\n'
    + '<div class="my-class"></div>'
    ;
});

describe('Dresscode', () =>
{
  it('should have no output if autoDress is disabled', () =>
  {
    expect(Dresscode.output).toMatchSnapshot();
  });

  it('should store correct CSS ruleset in Dresscode.output when Dresscode.dress() is called', () =>
  {
    Dresscode.dress();
    expect(Dresscode.output).toMatchSnapshot();
  });
});

