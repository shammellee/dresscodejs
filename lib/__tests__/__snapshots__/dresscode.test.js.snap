// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dresscode should have no output if autoDress is disabled 1`] = `undefined`;

exports[`Dresscode should store correct CSS ruleset in Dresscode.output when Dresscode.dress() is called 1`] = `".my-class{background-color:blue;color:yellow;font-size:2rem;padding:5px}"`;
